### PR TITLE
Implement passing object properties to decorators

### LINKED ISSUE #598: Responses other than 200 not being added to swagger gen.
## Sorting

* **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] support request

* I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior
With this method signature.

```ts
    @Response<ErrorResponse>(FORBIDDEN.status)
    @Response<ErrorResponse>(INVALID_REQUEST_SCHEMA.status)
    @Response<ErrorResponse>(INVALID_REGION.status)
    @Response<ErrorResponse>(INVALID_PASSWORD.status)
    @Post()
    public async postDevAccount(@Body() requestBody: DevAccountRequest): Promise<DevAccountResponse> {
```

I expected to get the ErrorResponse and usage of it in the swagger.json

## Current Behavior
Instead I get the error response generated but it not referenced anywhere in swagger.json
```json
"ErrorResponse": {
				"properties": {
					"code": {
						"type": "string"
					},
					"description": {
						"type": "string"
					},
					"status": {
						"type": "number",
						"format": "double"
					}
				},
				"required": [
					"code",
					"description",
					"status"
				],
				"type": "object",
				"additionalProperties": true
			},
```

```json
"/devaccount": {
			"post": {
				"operationId": "PostDevAccount",
				"responses": {
					"200": {
						"content": {
							"application/json": {
								"schema": {
									"$ref": "#/components/schemas/DevAccountResponse"
								}
							}
						},
						"description": "Ok"
					}
				},
```

## Steps to Reproduce
1. Write method signature like the above
2. run an npm script like "mkdir -p ./generated && tsoa swagger && tsoa routes"
3. observe that there is only one value in "response" for the method.

## Context (Environment)
Version of the library: 2.5.13
Version of NodeJS: v12.13.0
* Confirm you were using yarn not npm: [ ]

## Suggested solution

Provide more information in the README.md about how the @Response tag works assuming I'm just using it wrong.

### PR DESCRIPTION
**Pull request by dimitor115**
### All Submissions:
* [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
* [x] Have you written unit tests?
* [ ] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
* [x] This PR is associated with an existing issue?

I have extended the initial scope of this task, to enable handling object properties (`PropertyAccessExpression`) in all types of decorators. I have unified the logic of taking values from decorators.

**Closing issues**
closes #598 

**Test plan**
I know, I should extend tests to cover values pass as object properties in every decorator type, but first I want to grab the maintainer's opinion about the solution and idea.

### CODE REVIEW

**Comment by dimitor115** [comment]
@WoH please let me know, what do you think about this concept of unifying the decorator values handling :smile: 
Then I will add missing tests etc.

**Comment by WoH** [comment]
The approach looks good to me. I haven't had time to look at it in detail, but calling `getDecoratorValue` as often as possible is what I was hoping for.
Would be nice if you could also test using an Enum to pass a number.

**Comment by dimitor115** [comment]
@Woh so should I target for both branches, `master` and `3.x`?

**Comment by WoH** [comment]
> @WoH so should I target for both branches, `master` and `3.x`?

If you only want this in the 3.0-beta, you can target `3.x`.
If you want this in 2.5 and 3.0-beta, open a PR for both branches.
If you don't care about the 3.0-beta, target only master and I'll try to port this to the 3.0-beta, but I wanted to make sure you know the other branch.

**Comment by dimitor115** [comment]
@WoH yeah it ready for review. When we finish the review process, I will port this also to 3.X :smile:

File: src/metadataGeneration/initializer-value.ts [hunk 1]
```diff
@@ -43,10 +43,12 @@ export const getInitializerValue = (initializer?: ts.Expression, type?: Tsoa.Typ
...
       const nestedObject: any = {};
       objectLiteral.properties.forEach((p: any) => {
-        nestedObject[p.name.text] = getInitializerValue(p.initializer);
+        nestedObject[p.name.text] = getInitializerValue(p.initializer, typeChecker);
       });
       return nestedObject;
     default:
-      return;
+      const symbol = typeChecker.getSymbolAtLocation(initializer);
+      const extractedInitializer = symbol && symbol.valueDeclaration && (symbol.valueDeclaration as any).initializer;
```
Comment (WoH) [inline src/metadataGeneration/initializer-value.ts:51]: Let's avoid breaking the type checking with any here. Something like this: 
```suggestion
      const hasInitializer = (node: ts.Node) node is ts.HasInitializer => node.hasOwnProperty('initializer');
      const extractedInitializer = symbol && symbol.valueDeclaration && hasInitializer(symbol.valueDeclaration) && symbol.valueDeclaration.initializer;
```

Comment (dimitor115) [reply src/metadataGeneration/initializer-value.ts]: Ok, I have fix it :smile:

File: src/metadataGeneration/initializer-value.ts [hunk 2]
```diff
@@ -48,7 +48,9 @@ export const getInitializerValue = (initializer?: ts.Expression, typeChecker?: t
       return nestedObject;
     default:
       const symbol = typeChecker.getSymbolAtLocation(initializer);
-      const extractedInitializer = symbol && symbol.valueDeclaration && (symbol.valueDeclaration as any).initializer;
+      const extractedInitializer = symbol && symbol.valueDeclaration && hasInitializer(symbol.valueDeclaration) && (symbol.valueDeclaration.initializer as ts.Expression);
```
Comment (WoH) [inline src/metadataGeneration/initializer-value.ts:51]: Do we need the cast here?

**Comment by dimitor115** [comment]
@WoH Should I port similar solution for 3.X version?

**Comment by WoH** [comment]
Yes, would be appreciated a lot. You can target master for that, which now points to the 3.x branch.

**Pull request merged**